### ISSUE TITLE
Container cannot start with forwarded device and "required=false"

### ISSUE THREAD

**Issue by eraserix** [lxc/lxd#4495]
# Issue description

I tried to use the required=false option for device forwarding. But I cannot start the container once I set the flag to "false". The container is privileged and runs from the lxc 3.0 snap on Ubuntu 14.04.

# Steps to reproduce

```
$ lxc init images:debian/8 d8 -c security.privileged=true
Creating d8
$ lxc config device add d8 cpudma unix-char path=/dev/cpu_dma_latency required=false
Device cpudma added to d8

$ lxc start d8
Error: not a directory
Try `lxc info --show-log d8` for more info
```
lxc info --show-log d8 does not really help:

```
Name: d8
Remote: unix://
Architecture: x86_64
Created: 2018/04/25 12:59 UTC
Status: Stopped
Type: persistent
Profiles: default

Log:
```

**Comment by stgraber** [comment]
I'll look into this.

**Comment by eraserix** [comment]
Does this also fix the case for "disk" devices? Any chance to get this fixed in 3.0 as well?
